app/models/package_manager/pypi.rb

```diff
@@ -152,5 +152,14 @@ def self.project_find_names(project_name)
         project_name.gsub("_", "-"),
       ]
     end
+
+    # checks to see if the package exists on PyPI and the name matches the canonical name
+    def self.valid_project?(name)
+      raw_project = project(name)
+      mapped_project = mapping(raw_project)
```

Comment:
I think we might run into a case here with "status=Removed" projects where `raw_project` is nil, eg `astoptimizer`, and we don't necessarily want to delete them. Would it be weird to return `true` early here in those cases? Maybe a comment could help clarify why we'd do that too

Reply:
Updated to skip any projects that are labeled as removed in c808e9ecc4aa85a2d0666c094062561cf1f5e4f6